cachito/workers/pkg_managers/pip.py

```diff
@@ -1575,3 +1578,104 @@ def get_hosted_repositories_username(request_id):
     :rtype: str
     """
     return f"cachito-pip-{request_id}"
+
+
+def generate_pip_conf_content(nexus_pypi_hosted_repo_url, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_pypi_hosted_repo_url: the URL for the Nexus PyPI hosted repository
+        for the request
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://pip.pypa.io/en/stable/user_guide/#basic-authentication-credentials
+    # Create the request specific nexus repo url to be passed as `index-url` within pip.conf
+    if "://" in nexus_pypi_hosted_repo_url:
+        nexus_pypi_hosted_repo_url = nexus_pypi_hosted_repo_url.replace(
+            "://", f"://{username}:{password}@", 1
+        )
+    else:
+        raise CachitoError(
+            f"Nexus PyPI Hosted Repo URL: {nexus_pypi_hosted_repo_url} is not a valid URL"
+        )
+
+    pip_conf_content = textwrap.dedent(
+        f"""\
+        [global]
+        index-url = {nexus_pypi_hosted_repo_url}
+        """
+    )
+    if custom_ca_path:
+        # The CA could be embedded in the actual pip.conf file with the `cert` option, but this
+        # would mean that the CA file contents would be duplicated in the Cachito database for
+        # every request that uses a pip conf file
```

Comment:
Is this true? I know this works for NPM, but I haven't found any Pip documentation about this. If you have a source for this, it would be nice to include it, otherwise I would drop the comment.